README.md

```diff
@@ -211,6 +213,23 @@ $records = $porter->import(
         ->setCacheAdvice(CacheAdvice:SHOULD_CACHE())
 );
 ```
+### Cache key
+
+The cache key can optionally be generated by an implementation of `CacheKeyGeneratorInterface` passed into `CachingConnector::fetch`. This implementation should provide one method `generateCacheKey` which returns a [PSR-6][PSR-6]-compatible cache key.
+
+The default implementation `SourceAndOptionsHashCacheKeyGenerator` generates keys comprised of the source and options parameters passed to `Connector::fetch`. Options are sorted before the cache key is created so the order of options are insignificant.
+
+#### Implementation example
+
+```php
+class MyCacheKeyGenerator implements CacheKeyGenerator
+{
+    public function generateCacheKey($source, array $optionsSorted)
+    {
+        return str_replace(str_split('{}()/\@:'), '.', json_encode([$source, $optionsSorted], JSON_UNESCAPED_SLASHES));
```

Comment:
How do you feel about wrapping the `json_encode` example in  `md5`? With only a slight increase in complexity, we create a valid runnable example, without specifically pandering to PSR-6.